src/storage/disk_table.cc

```diff
@@ -206,7 +208,8 @@ bool DiskTable::Put(const std::string& pk, uint64_t time, const char* data, uint
     rocksdb::Status s;
     std::string combine_key = CombineKeyTs(pk, time);
     rocksdb::Slice spk = rocksdb::Slice(combine_key);
-    s = db_->Put(write_opts_, cf_hs_[1], spk, rocksdb::Slice(data, size));
+    auto cf_hs(cf_hs_);
```

Comment:
As we use `std::atomic_store_explicit` to update `cf_hs_`, we don't need to consider the data race like this.
`cf_hs` is either constructed from the old `cf_hs_` or the new `cf_hs_`.

Reply:
No. It's not like this. `std::atomic_store_explicit` must be used tother with `std::atomic_load_explicit` to guarantee the `atomicity`. If it is not true that you can use normal load if you use `std::atomic_store_explicit`.

See the note from cppreference:
`
To avoid data races, once a shared pointer is passed to any of these functions, it cannot be accessed non-atomically. In particular, you cannot dereference such a shared_ptr without first atomically loading it into another shared_ptr object, and then dereferencing through the second object.
`

https://en.cppreference.com/w/cpp/memory/shared_ptr/atomic

Reply:
Got it. My bad.
Thanks for the explanation